src/lib/maplibre/layer_helper.ts

```diff
@@ -0,0 +1,33 @@
+import type { Map } from "maplibre-gl";
+
+export class LayerHelper {
+  id: string;
+  defaultPaintValues: Array<[string, string]>;
+  changedValues: Array<string> = [];
+
+  constructor(layer: any) {
+    this.id = layer.id;
+    this.defaultPaintValues = this.getLayerValues(layer);
+  }
+
+  getLayerValues(layer: any): Array<[string, string]> {
+    return Object.keys(layer.paint ? layer.paint : {}).map(
```

Comment:
Why not just `Object.entries` and maybe deep-cloning for sanity? That gives key/values outright